ttnn/cpp/ttnn/tensor/xtensor/partition.hpp

```diff
@@ -0,0 +1,27 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/tensor/xtensor/xtensor_all_includes.hpp"
+
+namespace ttnn::experimental::xtensor {
+
+// IMPORTANT: `chunk` and `concatenate` are not yet part of the public ttnn API. Internally, they rely on the `xtensor`
+// library for efficient host-side operations.
+
+// Splits a tensor into chunks along the specified dimension.
+std::vector<tt::tt_metal::Tensor> chunk(const tt::tt_metal::Tensor& tensor, int num_chunks, int dim = 0);
```

Comment:
@sminakov-tt FYI since we talked about this yesterday. This is one of the primitives used to build the user-facing multi-device sharding (this adds C++ parity now).